File: api/src/main/java/com/ruby/rubia_server/core/entity/Conversation.java
// src/main/java/com/ruby/rubia_server/core/entity/Conversation.java
package com.ruby.rubia_server.core.entity;

import com.ruby.rubia_server.core.enums.Channel;
import com.ruby.rubia_server.core.enums.ConversationStatus;
import com.ruby.rubia_server.core.enums.ConversationType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "conversations")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Conversation {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id", nullable = false)
    private Company company;

    @Enumerated(EnumType.ORDINAL)
    @Column(name = "channel", nullable = false)
    private Channel channel; // WHATSAPP, etc.

    @Enumerated(EnumType.ORDINAL)
    @Column(name = "status", nullable = false)
    private ConversationStatus status; // ENTRADA, ESPERANDO, FINALIZADOS

    @Column(name = "priority")
    private Integer priority; // 1-5, etc.

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assigned_user_id")
    private User assignedUser; // Usuário humano atribuído a esta conversa (pode ser nulo para grupos ou não atribuídas)

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_user_id")
    private User ownerUser; // Usuário humano que criou/é o "dono" da conversa (pode ser nulo)

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "campaign_id")
    private Campaign campaign; // Opcional, para indicar se a conversa faz parte de uma campanha

    // NOVO CAMPO: Tipo da conversa (individual ou grupo)
    @Enumerated(EnumType.ORDINAL)
    @Column(name = "conversation_type", nullable = false)
    @Builder.Default
    private ConversationType conversationType = ConversationType.ONE_TO_ONE;

    // Campo para vincular com o chatLid da Z-API
    @Column(name = "chat_lid", unique = true)
    private String chatLid;

    // Remove a relação direta com Customer.
    // Os participantes (incluindo Customers) são agora gerenciados pela entidade ConversationParticipant.
    @OneToMany(mappedBy = "conversation", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    @Builder.Default
    private List<ConversationParticipant> participants = new ArrayList<>();

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: api-dotnet/Rubia.Server/Entities/Conversation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Rubia.Server.Enums;

namespace Rubia.Server.Entities;

[Table("conversations")]
public class Conversation : BaseEntity
{
    [Column("channel")]
    [Required]
    public Channel Channel { get; set; } // WHATSAPP, etc.

    [Column("status")]
    [Required]
    public ConversationStatus Status { get; set; } // ENTRADA, ESPERANDO, FINALIZADOS

    [Column("priority")]
    public int? Priority { get; set; } // 1-5, etc.

    [Column("conversation_type")]
    [Required]
    public ConversationType ConversationType { get; set; } = ConversationType.OneToOne;

    [Column("chat_lid")]
    public string? ChatLid { get; set; }

    [Column("ai_auto_response_enabled")]
    [Required]
    public bool AiAutoResponseEnabled { get; set; } = true;

    [Column("ai_messages_used")]
    [Required]
    public int AiMessagesUsed { get; set; } = 0;

    [Column("ai_limit_reached_at")]
    public DateTime? AiLimitReachedAt { get; set; }

    // Navigation properties
    [Column("company_id")]
    [Required]
    public Guid CompanyId { get; set; }
    
    [ForeignKey("CompanyId")]
    public virtual Company Company { get; set; } = null!;

    [Column("assigned_user_id")]
    public Guid? AssignedUserId { get; set; }
    
    [ForeignKey("AssignedUserId")]
    public virtual User? AssignedUser { get; set; } // Usuário humano atribuído a esta conversa

    [Column("owner_user_id")]
    public Guid? OwnerUserId { get; set; }
    
    [ForeignKey("OwnerUserId")]
    public virtual User? OwnerUser { get; set; } // Usuário humano que criou/é o "dono" da conversa

    [Column("campaign_id")]
    public Guid? CampaignId { get; set; }
    
    [ForeignKey("CampaignId")]
    public virtual Campaign? Campaign { get; set; } // Opcional, para indicar se a conversa faz parte de uma campanha

    public virtual ICollection<ConversationParticipant> Participants { get; set; } = new List<ConversationParticipant>();
    public virtual ICollection<Message> Messages { get; set; } = new List<Message>();
}